### DISCUSSION TITLE
Is it possible for NumberInput to have a null or undefined value when it's not a required field?

### DISCUSSION THREAD

**Discussion by TrashPandacoot** [mantinedev/mantine#356] [Q&A] [answered]
Basically in my use case I have a decent amount of number fields that are optional fields that don't have a natural default value. They're usually things that will be entered at a later date by editing an existing record so currently if one of these optional number fields is left blank it's "we don't know what the value is going to be at this time". 

I took a peak at the NumberInput source and I'm pretty sure this behavior I'm describing is not possible with the current implementation. I'm guessing just based on the `onChange` signature it'd be a breaking change in order to do though.

Is adding this sort of behavior something that makes sense for this library? I'd be happy to contribute if it is, although I'd likely need a bit of guidance as I am a little lost trying to follow the current implementation

**Comment by rtivital** [comment, marked as answer]
Hi @TrashPandacoot it would be a great feature to contribute, please follow contribution guide – https://mantine.dev/pages/contributing/ and reach out on Discord in **Contributing** channel if you need any help
